cg/meta/workflow/fastq.py

```diff
@@ -0,0 +1,275 @@
+"""
+This module handles concatenation of balsamic fastq files.
+
+Classes:
+    FastqFileNameCreator: Creates valid balsamic filenames
+    FastqHandler: Handles fastq file linking
+"""
+import datetime as dt
+import gzip
+import logging
+import os
+import re
+import shutil
+from pathlib import Path
+from typing import List, Optional
+
+LOG = logging.getLogger(__name__)
+
+DEFAULT_DATE_STR = (
+    "171015"  # Stand in value to use if sequencing date cannot be extracted from header
+)
+DEFAULT_INDEX = (
+    "XXXXXX"  # Stand in value to use if flowcell index is to be masked when renaming file
+)
+
+
+class FastqHandler:
+    """Handles fastq file linking"""
+
+    @staticmethod
+    def concatenate(files: List, concat_file: str):
+        """Concatenates a list of fastq files"""
+        LOG.info(FastqHandler.display_files(files, concat_file))
+
+        with open(concat_file, "wb") as wfd:
+            for fil in files:
+                with open(fil, "rb") as file_descriptor:
+                    shutil.copyfileobj(file_descriptor, wfd)
+
+        size_before = FastqHandler.size_before(files)
+        size_after = FastqHandler.size_after(concat_file)
+
+        try:
+            FastqHandler.assert_file_sizes(size_before, size_after)
+        except AssertionError as error:
+            LOG.warning(error)
+
+    @staticmethod
+    def size_before(files: List) -> int:
+        """Returns the total size of the linked fastq files before concatenation"""
+
+        return sum(os.stat(f).st_size for f in files)
+
+    @staticmethod
+    def size_after(concat_file: str) -> int:
+        """returns the size of the concatenated fastq files"""
+
+        return os.stat(concat_file).st_size
+
+    @staticmethod
+    def assert_file_sizes(size_before: int, size_after: int) -> None:
+        """asserts the file sizes before and after concatenation. If the file sizes differ by more
+        than 1 percent, throw an exception"""
+        msg = (
+            f"Warning: Large file size difference after concatenation!"
+            f"Before: {size_before} -> after: {size_after}"
+        )
+
+        assert abs(size_before - size_after) / size_before <= 0.01, msg
```

Comment:
I don't get this, why is there a comma and msg in the assert?

Reply:
The message is to be displayed should assertion error be raised...